pilot/pkg/serviceregistry/external/servicediscovery.go

```diff
@@ -132,6 +133,12 @@ func (d *ServiceEntryStore) GetService(hostname model.Hostname) (*model.Service,
 	return nil, nil
 }
 
+// GetServiceNamespace retrieves namespace of a service if it exists. Currently it's not implemented
+// and only returns empty string.
+func (d *ServiceEntryStore) GetServiceNamespace(service *model.Service) (string, error) {
+	return "", fmt.Errorf("NOT IMPLEMENTED")
+}
```

Comment:
```go
for _, config := range d.store.ServiceEntries() {
 		serviceEntry := config.Spec.(*networking.ServiceEntry)
 		svcs := convertServices(serviceEntry)
                 for _, s := range svcs {
                   if s.Hostname == service.Hostname {
                               return config.Meta.namespace, nil
                  }
               }
          return "", fmt.Errorf("service not found")
 	}
```

Reply:
I don't think config namespace is the right value here... Is there some restriction on placing hosts into config namespaces?

Reply:
No. But google.com does not belong in any arbitrary namespace either. This is a temporary change

Reply:
Is there an issue that config for virtual service that references a k8s service is in the namespace of k8s service? There is a multitude of downstream systems that depend on that assumption...